### PR TITLE
Roomsettings: Support a per-room default mod for /dt

### PR DESCRIPTION
**Pull request by mia-pi-git**
Useful utility.

### CODE REVIEW

**Comment by Zarel** [comment]
Don't rooms already have a default format for other commands? Did we never implement that?

**Comment by Zarel** [comment]
I think a default format makes more sense than a default mod.

File: server/rooms.ts [hunk 1]
```diff
@@ -118,6 +118,7 @@ export interface RoomSettings {
 	repeats?: RepeatedPhrase[];
 	autoModchat?: {rank: GroupSymbol, time: number, active: boolean};
 	tournaments?: TournamentRoomSettings;
+	detailsMod?: string;
```
Comment (Zarel) [inline server/rooms.ts:121]: I would call this `defaultFormat` and plan for it to be eventually used for all relevant commands, such as `/learn`.

**Comment by mia-pi-git** [comment]
> Don't rooms already have a default format for other commands? Did we never implement that?

No, we did not. You may be thinking of data command tier display? But that's just singles / doubles.

**Comment by Zarel** [comment]
Like I said in DMs, this should be `defaultFormat`.

**Comment by xInstruct** [comment]
> Don't rooms already have a default format for other commands? Did we never implement that?

afaik the only thing that rooms can edit /dt is the smogon tier (e.g. LC room has it to use doubles tiers as opposed to standard singles)

**Comment by Zarel** [comment]
(I'm working on this now)

**Comment by DaWoblefet** [comment]
This was superseded by https://github.com/smogon/pokemon-showdown/pull/8237

**Pull request closed without merging**